ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ExecutionContext.java

```diff
@@ -18,15 +15,10 @@
  * A batch request is that in neural-search side multiple fields are send in one request to ml-commons,
  * but the remote model doesn't accept list of string inputs so in ml-commons the request needs split.
  * sequence is used to identify the index of the split request.
- * countDownLatch is used to wait for all the split requests to finish.
- * exceptionHolder is used to hold any exception thrown in a split-batch request.
  */
 @Data
 @AllArgsConstructor
 public class ExecutionContext {
     // Should never be null
     private int sequence;
-    private CountDownLatch countDownLatch;
```

Comment:
Why remove this part ? Is it ok to remove this @zane-neo, can you confirm ?

Reply:
This PR is not purely adding retry, it also contains code refactoring. 

I see several times that people want to fix other issues/add some other minor changes in same PR, but it turns out the extra change brings new bugs. Please work with Zan to make sure it will not break BWC.

Reply:
This is a neccessary refactor to support retry for split-batch.

Reply:
Sure, I've gone through several meetings with Zan about this changes. And also conducted experiments at my environment to test these code.

I'll follow up if @zane-neo has any concerns on this

Reply:
Since using GroupActionListener and this listener internally handles the multiple split request's response so no need to handle this with CountDownLatch, this change LGTM.